library/src/main/java/com/mapzen/android/OverlayManager.java

```diff
@@ -183,6 +207,146 @@ public MapData addMarker(Marker marker) {
         return addPointToMarkerMapData(marker);
     }
 
+    /**
+     * Draws two pins on the map. The start pin is active and the end pin is inactive.
+     * @param start
+     * @param end
+     */
+    public void drawRoutePins(LngLat start, LngLat end) {
+        if (startPinData == null) {
+            startPinData = mapController.addDataLayer(NAME_START_PIN);
+        }
+        if (endPinData == null) {
+            endPinData = mapController.addDataLayer(NAME_END_PIN);
+        }
+        startPinData.addPoint(start, null);
+        endPinData.addPoint(end, null);
+        mapController.requestRender();
```

Comment:
Need to verify with @blair1618 or @tallytalwar but I don't think calling `mapController.requestRender()` is required any longer when adding client data to the map. I tried commenting these out and everything seems to still work.